From d512061ff457004bb0b19b3be01e110d1e183b54 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Wed, 22 Feb 2017 08:26:38 -0600
Subject: [PATCH] netd: fix tethering

-This command fails but Android 6.0 iptables was IPV4 anyways
I think. So this really doesn't matter to us..

Change-Id: I5e858e912abfd36d39893870a2ffa05cb18f9dfd
---
 server/NatController.cpp | 55 ++++++++++++++++++++++++------------------------
 1 file changed, 28 insertions(+), 27 deletions(-)

diff --git a/server/NatController.cpp b/server/NatController.cpp
index cda8f5f..c25f10d 100644
--- a/server/NatController.cpp
+++ b/server/NatController.cpp
@@ -148,7 +148,6 @@ int NatController::setDefaults() {
 
 int NatController::enableNat(const char* intIface, const char* extIface) {
     ALOGV("enableNat(intIface=<%s>, extIface=<%s>)",intIface, extIface);
-
     if (!isIfaceName(intIface) || !isIfaceName(extIface)) {
         errno = ENODEV;
         return -1;
@@ -345,25 +344,27 @@ int NatController::setForwardRules(bool add, const char *intIface, const char *e
             LOCAL_TETHER_COUNTERS_CHAIN
     };
 
-    const char *cmd4[] = {
-            IP6TABLES_PATH,
-            "-w",
-            "-t",
-            "raw",
-            add ? "-A" : "-D",
-            LOCAL_RAW_PREROUTING,
-            "-i",
-            intIface,
-            "-m",
-            "rpfilter",
-            "--invert",
-            "!",
-            "-s",
-            "fe80::/64",
-            "-j",
-            "DROP"
-    };
-
+// Meticulus: avoid Werror by not creating this cmd.
+// Comment long way so we can see in patch.
+//    const char *cmd4[] = {
+//            IP6TABLES_PATH,
+//            "-w",
+//            "-t",
+//            "raw",
+//            add ? "-A" : "-D",
+//            LOCAL_RAW_PREROUTING,
+//            "-i",
+//            intIface,
+//            "-m",
+//            "rpfilter",
+//            "--invert",
+//            "!",
+//            "-s",
+//            "fe80::/64",
+//            "-j",
+//            "DROP"
+//    };
+//
     if (runCmd(ARRAY_SIZE(cmd2), cmd2) && add) {
         // bail on error, but only if adding
         rc = -1;
@@ -376,10 +377,10 @@ int NatController::setForwardRules(bool add, const char *intIface, const char *e
         goto err_return;
     }
 
-    if (runCmd(ARRAY_SIZE(cmd4), cmd4) && add) {
-        rc = -1;
-        goto err_rpfilter;
-    }
+//    if (runCmd(ARRAY_SIZE(cmd4), cmd4) && add) {
+//        rc = -1;
+//        goto err_rpfilter;
+//    }
 
     if (setTetherCountingRules(add, intIface, extIface) && add) {
         rc = -1;
@@ -388,9 +389,9 @@ int NatController::setForwardRules(bool add, const char *intIface, const char *e
 
     return 0;
 
-err_rpfilter:
-    cmd3[2] = "-D";
-    runCmd(ARRAY_SIZE(cmd3), cmd3);
+//err_rpfilter:
+//    cmd3[2] = "-D";
+//    runCmd(ARRAY_SIZE(cmd3), cmd3);
 err_return:
     cmd2[2] = "-D";
     runCmd(ARRAY_SIZE(cmd2), cmd2);

